src/init/VarResolver.ts

```diff
@@ -0,0 +1,139 @@
+import yargs from 'yargs';
+import { AsyncHandler } from '../util/handlers/AsyncHandler';
+import { ensureTrailingSlash, modulePathPlaceholder, resolveAssetPath } from '../util/PathUtil';
+
+const defaultConfig = `${modulePathPlaceholder}config/default.json`;
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+/**
+ * CLI options needed for performing meta-process of app initialization.
+ * These options doesn't contribute to components-js vars normally.
+ */
+export const baseYargsOptions: Record<string, yargs.Options> = {
+  config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+  loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true },
+  mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
+  varConfig: { type: 'string', alias: 'v', default: defaultVarConfig, requiresArg: true },
+};
+
+/**
+ * A handler that takes args, and returns computed variable value
+ */
+export abstract class VarComputer extends AsyncHandler<yargs.Arguments, any> {
+  public abstract handle(args: yargs.Arguments): Promise<any>;
+}
+
+// Currently componentsjs-builder giving error, if we use external TSTypeReference.
+// Re-definition is temporary work around.
+export interface IYargsOptions extends yargs.Options {
+  alias?: string | readonly string[];
+  default?: any;
+  desc?: string;
+  requiresArg?: boolean;
+  type?: 'array' | 'count' | yargs.PositionalOptionsType;
+}
+
+export interface IVarResolverOptions {
+  yargsOptions: Record<string, IYargsOptions>;
+  varComputers: Record<string, VarComputer>;
+  usage?: string;
+  strictMode?: boolean;
+}
+
+export class VarResolver {
```

Comment:
```suggestion
export class VarResolver extends AsyncHandler<string[], Record<string, unknown>> {
```

Since this class has only 1 function that it exposes (`resolveVars`), we could make this an `AsyncHandler` where the handle function does what the `resolveVars` function does now.

This does not have an immediate use right now, but we have several utility classes for `AsyncHandler`s so this might help somewhere in the future if we ever want to extend the behaviour here.